### ISSUE TITLE
Enhancement of upload profile image 

### ISSUE THREAD

**Issue by shilpashingnapure** [Innovateninjas/Paws-frontend#348]
1. refer this issue #243
2. Adding the upload profile image at time of registrations for both user

**Comment by shilpashingnapure** [comment]
@codewarnab or @rishicds can you please add label for this issue

**Comment by rishicds** [comment]
Best of luck @shilpashingnapure

**Comment by codewarnab** [comment]
@shilpashingnapure are you planning to add it in the registration page itself ? 
i think it will be better if we ask the user immediately after the registartion we need to keep as much simple as possible

**Comment by codewarnab** [comment]
also try to make the profile picture is an optional thing probably adding a random avtar would be enough and user can change it if they want

**Comment by shilpashingnapure** [comment]
> @shilpashingnapure are you planning to add it in the registration page itself ? i think it will be better if we ask the user immediately after the registration we need to keep as much simple as possible

I don't get it this. you mean once registration is done after that we have ask user if you want to upload profile pic or not like this ?

**Comment by shilpashingnapure** [comment]
I guess this feature will get complicate then , for simplicity then we can just keep upload  photo option for profile page  and also we are showing default image to users so i think we don't need to add this what you think guys ?

**Comment by codewarnab** [comment]
> > @shilpashingnapure are you planning to add it in the registration page itself ? i think it will be better if we ask the user immediately after the registration we need to keep as much simple as possible
> 
> I don't get it this. you mean once registration is done after that we have ask user if you want to upload profile pic or not like this ?
yes

**Comment by codewarnab** [comment]
> I guess this feature will get complicate then , for simplicity then we can just keep upload photo option for profile page only and also we are already showing default image to users in profile page so i think we don't need to add this what you think guys ?

yeah it might be get complicated . but our main priority is the easiest possible reporting and if any user want to register it should be super simple 

i think showing a modal if they want to upload profile picture or not would be good then create a uploading section their or just redirect to profile page and show a tooltip for uplloading profile picture .

let me know what are your thoughts on this

**Comment by shilpashingnapure** [comment]
this will be good idea , after registration we can redirect to profile page and show a tooltip for uploading

**Comment by codewarnab** [comment]
> this will be good idea , after registration we can redirect to profile page and show a tooltip for uploading

as it is complicated task it is better to break it in multiple issues  . you can do it else i will do it tonight

**Comment by shilpashingnapure** [comment]
now we are redirecting to the page , so i think we no need of multiple issues , we can do this in single task only.

**Comment by Tani1126** [comment]
pls, assign this issue to me, i am a contributor at GSSOC'24.

**Comment by codewarnab** [comment]
@shilpashingnapure  hey i was just checking the paws you have added the profile picture functionality before why is the feature is coming when i click on the profile icon in profile page ?

**Comment by rishicds** [comment]
yes @codewarnab , check in a few mins

**Comment by shilpashingnapure** [comment]
> @shilpashingnapure hey i was just checking the paws you have added the profile picture functionality before why is the feature is coming when i click on the profile icon in profile page ?
can you show me in video.. so i get clear what happened

**Comment by codewarnab** [comment]
> > @shilpashingnapure hey i was just checking the paws you have added the profile picture functionality before why is the feature is coming when i click on the profile icon in profile page ?
> > can you show me in video.. so i get clear what happened

nope its working fine

**Comment by codewarnab** [comment]
> now we are redirecting to the page , so i think we no need of multiple issues , we can do this in single task only.

yeah but after registration you need show a modal that if user wants to upload profile pic or not if not then dont redirect else redirect it will not much hard as already their are two modals in the app you can just take the code from their so if you wan this work under one issue feel free to start working on it also we have updated the deployed branch you can test the profile picture functionality properly if you find any bugs

**Comment by shilpashingnapure** [comment]
Hi , can you assign me and label for this close PR https://github.com/Innovateninjas/Paws-frontend/pull/344
other wise , my point will not get count. (don't know now it will counted or not , but still if possible can you assign me).
actually today I got mail , in there its mention in PR also need to be assigned and label.

**Comment by rishicds** [comment]
@shilpashingnapure that pr and issue from backend repository will be transferred here and will be assigned the appropriate labels, then it will be counted :)

**Comment by shilpashingnapure** [comment]
> @shilpashingnapure that pr and issue from backend repository will be transferred here and will be assigned the appropriate labels, then it will be counted :)

for this two backend PR which merged , if possible can you assign https://github.com/Innovateninjas/Paws-Backend/pull/11 , https://github.com/Innovateninjas/Paws-Backend/pull/10

And Thank You so Much for clearing my doubt 😊.

**Comment by shilpashingnapure** [comment]
@codewarnab  for this  backend issue , you only added gssoc label , points label is not added , does it will count 😅?

**Comment by shilpashingnapure** [comment]
@codewarnab hi can you suggest me does this flow make sense ? 
for redirect I used URL with query params
and when we click on skip it will redirect to home page 

https://github.com/Innovateninjas/Paws-frontend/assets/69167064/22f7b4cb-9c7e-4010-aa64-d9dc738726a7

**Comment by codewarnab** [comment]
> @codewarnab hi can you suggest me does this flow make sense ? for redirect I used URL with query params and when we click on skip it will redirect to home page
> 
>  video1108293007.mp4

Yes looking perfect

**Comment by shilpashingnapure** [comment]
@codewarnab Hey , today I tested the upload feature , but now its not working then i got to know in backend has some changes.
In frontend I need to fixed this , for this I have to created new issue right?
and also when i see in backend its have profile update info , so we have to update profile info also ( like name and phone number) ? 
because in frontend we are only able to update profile photo only.

**Comment by codewarnab** [comment]
> @codewarnab Hey , today I tested the upload feature , but now its not working then i got to know in backend has some changes. In frontend I need to fixed this , for this I have to created new issue right? and also when i see in backend its have profile update info , so we have to update profile info also ( like name and phone number) ? because in frontend we are only able to update profile photo only.

what fix needed in the frontend anirban has given [commit](https://github.com/Innovateninjas/Paws-frontend/commit/79f9b97393822af1be2ea030bfd4ff6631bbf014) see if this fix the issue you taliking about
i think you can update any each them separately their will be no issue i guess as anirban stated this  [here](https://github.com/Innovateninjas/Paws-frontend/pull/286#discussion_r1614671839)

**Comment by shilpashingnapure** [comment]
okay , its already done. I have not seen commit of frontend my bad.
